File: ProyectoBiblioteca-back/ProyectoBibliotecaWeb/src/main/java/com/example/service/UsuarioService.java
package com.example.service;

import java.util.Map;

import org.springframework.http.ResponseEntity;

import com.example.model.Usuario;


public interface UsuarioService {

	public ResponseEntity<Map<String, Object>> listarUsuarios();
	public ResponseEntity<Map<String, Object>> guardarUsuario(Usuario usuario);
	public ResponseEntity<Map<String,Object>> eliminarUsuario(Long id);
	public ResponseEntity<Map<String,Object>> actualizarUsuario(Long id, Usuario usuario);
	public ResponseEntity<Map<String,Object>> obtenerUsuario(Long id);

}

File: ProyectoBiblioteca-front/src/app/servicio/usuario.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class UsuarioService {

  private baseUrl = 'http://localhost:8080/api/usuarios';

  constructor(private http: HttpClient) {}

  // Obtener perfil del usuario autenticado
  obtenerPerfil(): Observable<any> {
    return this.http.get(`${this.baseUrl}/perfil`);
  }

  // Listar todos los usuarios
  listarUsuarios(): Observable<any> {
    return this.http.get(`${this.baseUrl}/listar`);
  }

  // Obtener usuario por ID
  obtenerUsuario(id: number): Observable<any> {
    return this.http.get(`${this.baseUrl}/${id}`);
  }

  // Crear nuevo usuario
  crearUsuario(usuario: any): Observable<any> {
    return this.http.post(`${this.baseUrl}/crear`, usuario);
  }

  // Actualizar usuario existente
  actualizarUsuario(id: number, usuario: any): Observable<any> {
    return this.http.put(`${this.baseUrl}/${id}`, usuario);
  }

  // Eliminar usuario
  eliminarUsuario(id: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/${id}`);
  }
}